### PR TITLE
[cni/cilium] Add support for additional config fields

### PR DESCRIPTION
**Pull request by dntosas**
In this commit, we enable users to define their setup with following
additional fields:

- DisableEndpointCRD
- EnableEndpointHealthChecking
- IdentityAllocationMode
- IdentityChangeGracePeriod
- BPFLBAlgorithm
- BPFLBMaglevTableSize
- BPFNATGlobalMax
- BPFNeighGlobalMax
- BPFPolicyMapMax

Added also validation tests to prevent conflicting value combinations to
reach actual cluster state.

- [x] Add additional config fields
- [x] Add default values
- [x] Include validation/sanity checks on new fields and combinations

Signed-off-by: dntosas <ntosas@gmail.com>

### CODE REVIEW

File: pkg/apis/kops/networking.go [hunk 1]
```diff
@@ -324,6 +330,12 @@ type CiliumNetworkingSpec struct {
 	// EnvoyLog is not implemented and may be removed in the future.
 	// Setting this has no effect.
 	EnvoyLog string `json:"envoyLog,omitempty"`
+	// IdentityAllocationMode specifies  how identities ("crd", "kvstore").
```
Comment (olemarkus) [inline pkg/apis/kops/networking.go:333]: Cut sentence?

File: pkg/apis/kops/networking.go [hunk 2]
```diff
@@ -313,6 +316,9 @@ type CiliumNetworkingSpec struct {
 	// "never": Cilium allows all traffic regardless of policies in place.
 	// If unspecified, "default" policy mode will be used.
 	EnablePolicy string `json:"enablePolicy,omitempty"`
+	// EnableEndpointHealthChecking enables connectivity health checking between virtual endpoints.
+	// Default: true
+	EnableEndpointHealthChecking bool `json:"enableEndpointHealthChecking,omitempty"`
```
Comment (olemarkus) [inline pkg/apis/kops/networking.go:321]: How can default be true if this one is not a pointer?

File: pkg/apis/kops/networking.go [hunk 3]
```diff
@@ -434,6 +446,24 @@ type CiliumNetworkingSpec struct {
 	// BPFCTGlobalAnyMax is the maximum number of entries in the non-TCP CT table.
 	// Default: 262144
 	BPFCTGlobalAnyMax int `json:"bpfCTGlobalAnyMax,omitempty"`
+	// BPFLBAcceleration is the load balancing acceleration mode via XDP ("native", "disabled")
```
Comment (olemarkus) [inline pkg/apis/kops/networking.go:449]: This one isn't available on any supported distros. It require kernel 5.6+.

File: pkg/apis/kops/networking.go (hunk 1 above)
Comment (dntosas) [reply pkg/apis/kops/networking.go]: oops, good catch ^ fixed

File: pkg/apis/kops/networking.go (hunk 2 above)
Comment (dntosas) [reply pkg/apis/kops/networking.go]: true ^^ switched to pointer

File: pkg/apis/kops/networking.go (hunk 3 above)
Comment (dntosas) [reply pkg/apis/kops/networking.go]: aha.. removed ^^

File: pkg/apis/kops/networking.go [hunk 4]
```diff
@@ -434,6 +446,24 @@ type CiliumNetworkingSpec struct {
 	// BPFCTGlobalAnyMax is the maximum number of entries in the non-TCP CT table.
 	// Default: 262144
 	BPFCTGlobalAnyMax int `json:"bpfCTGlobalAnyMax,omitempty"`
+	// BPFLBAlgorithm is the load balancing algorithm ("random", "maglev")
```
Comment (olemarkus) [inline pkg/apis/kops/networking.go:449]: Nit: these are missing punctuations.

File: pkg/apis/kops/v1alpha2/networking.go [hunk 5]
```diff
@@ -254,6 +254,8 @@ type AmazonVPCNetworkingSpec struct {
 	Env []EnvVar `json:"env,omitempty"`
 }
 
+const CiliumIpamEni = "eni"
```
Comment (olemarkus) [inline pkg/apis/kops/v1alpha2/networking.go:257]: Maybe define a string type/enum for IPAMs?

**Review by olemarkus — CHANGES_REQUESTED** [review]
Thanks for this. Just some small nits.

File: pkg/apis/kops/networking.go (hunk 4 above)
Comment (dntosas) [reply pkg/apis/kops/networking.go:449]: could you suggest a change because I can't understand what's missing? :stuck_out_tongue:

Comment (olemarkus) [reply pkg/apis/kops/networking.go]: A period (.) at the end of the sentence :)

**Comment by dntosas** [comment]
lol, i have no idea how i manage to close this :thinking:

**Comment by olemarkus** [comment]
You pushed an empty branch, I think. when there is no diff, it auto-closes.

**Pull request closed without merging**